homeassistant/components/oralb/__init__.py

```diff
@@ -35,6 +36,11 @@ def _needs_poll(
     ) -> bool:
         # Only poll if hass is running, we need to poll,
```

Comment:
We shouldn't look at the registry here


Let's subclass the entities and coordinator and have them register for needing an active connection. Then we can decide to poll based on how many are registered that want an active connection

Reply:
What do you mean by registering that we want an active connection? Is that just the mode attribute forActiveBluetoothProcessorCoordinator? Or is it more to it than that?

Reply:
```python

class OralbActiveBluetoothProcessorCoordinator(ActiveBluetoothProcessorCoordinator):

    def __int__(self,.....)
         super().__init__(....)
         self._active_subscriptions = []

    def register_active(self, sub):
         self._active_subscriptions.append(sub)          

         def _unregister()
           self._active_subscriptions.remove(sub)

        return _unregister
    
     @property
     def needs_active_update(self):
          return bool(self._active_subscriptions)

```

Than subclass the entity as well to call `self.async_on_remove(self.coordinator.processor.register_active(sub))` in `async_added_to_hass`

Than you can check if we should pool if `needs_active_update`

Reply:
Alternatively change `hass.data[DOMAIN]` to be a dataclass ex https://github.com/home-assistant/core/blob/8a2d81fe4b1bff8df16a4b41657299d4b7481b45/homeassistant/components/lookin/models.py#L19 https://github.com/home-assistant/core/blob/c1039fab4a3978eda08eac17eb4aaf773b69b929/homeassistant/components/lookin/__init__.py#L159 that stores a new object and the `ActiveBluetoothProcessorCoordinator` and refactor `_needs_poll` and `_async_poll` to be bound methods to a new object that could keep track of when entities need active polling.

Reply:
@bdraco I implemented this, added a few comments I would appreciate you look at (no rush).

But it seems like the coordinator doesn't do the normal manufacturer data updates now, I'm sure I forgot to add something to the class or I made some mistake there, but if you know what it is, just let me know.

Thank you as always!